### ISSUE TITLE
Cannot join private/locked rooms

### ISSUE THREAD

**Issue by KingKarrow** [callaa/Drawpile#239]
I'm on OS X Yosemite (10.10), Drawpile 1.0.1

I go to Session > Join > Find (drawpile.net listing), and whenever I try to join a locked room it opens up the new session window, and a dialog box drops down that I can't interact with or cancel. The status bar on the bottom of the window says "Connecting to [IP]..."

Public rooms work fine.

Attached a screenshot, after selecting (for example) the Drawpile After Dark room.

![screenshot 2015-08-11 23 39 30](https://cloud.githubusercontent.com/assets/13758961/9216223/8f4dd818-4083-11e5-81b3-692e7b78a5e4.png)

**Comment by callaa** [comment]
That's weird. It should be popping up a password dialog. I have the same OSX version, but can't reproduce the bug myself.

**Comment by KingKarrow** [comment]
I figured that was the expected response. Is the password dialogue supposed to be where that blank dialog is?

**Comment by KingKarrow** [comment]
Oh! Okay, I got it to work, but only if I quit, then try joining with the IP that's been saved from the last connection attempt. I think the problem has something to do with how DP's host address' are passed in after you select them from the listing. For example, if I pick DP's AD, it puts this in this Host address field: "drawpile://88.193.119.127/dpad." This causes the next dialog to not function correctly. If I try again, but with just the IP address that's already in there (88.193.119.127), the next dialog gives me the list of DP's servers. I choose DP's AD from there, and THEN it gets to the password. 

So basically, I think the address it uses to try and bypass the next dialog ("choose server") just causes the next dialog to get confused over what it should be displaying.

**Comment by callaa** [comment]
I just managed to reproduce the problem myself! It happens on Linux too, so it's clearly a bug in Drawpile itself, rather than any platform integration issue.
The bug is triggered when there is more than one session running on a server and you try to connect directly to one that is password protected.
For some reason, the client pops up the in-server session selector, even though the session ID was specified in the URL, and the password query dialog at the same time. The password dialog has window-modal focus, but is hidden behind the session selector, leading to a locked up UI.

**Comment by callaa** [comment]
This is now fixed and will be included in the next release. Turned out to be quite simple: just needed to hide the session selector dialog when the password popup is shown!

**Comment by KingKarrow** [comment]
Nice! :)
